src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/NullTestAroundInstanceOfCheck.java

```diff
@@ -0,0 +1,232 @@
+                log(ast, MSG_KEY);
+            }
+        }
+    }
+
+    /**
+     * To get instanceOfNode or notEqualNode's full ident text.
+     *
+     * @param ast DetailAST
+     * @return string ident node's text
+     */
+    private static String getFullText(DetailAST ast) {
+        final DetailAST dotNode = ast.findFirstToken(TokenTypes.DOT);
+        final String fullText;
+        if (dotNode == null) {
+            fullText = ast.findFirstToken(TokenTypes.IDENT).getText();
+        }
+        else {
+            final FullIdent fullIdent = FullIdent.createFullIdent(dotNode);
+            fullText = fullIdent.getText();
+        }
+        return fullText;
+    }
+
+    /**
+     * Check if there is a unnecessary 'null' check before 'instanceof' expression when
+     * 'instanceof' expression is on next line of 'null' check and vice versa.
+     *
+     * @param ast DetailAST
+     * @return error boolean
+     */
+    private static boolean checkOnNextLine(DetailAST ast) {
+        final boolean error1 = checkOuterNotEqual(ast);
+        final boolean error2 = checkOuterInstanceOf(ast);
+        return error1 || error2;
+    }
+
+    private static boolean checkOuterNotEqual(DetailAST ast) {
+        final DetailAST outerNotEqualNode = getExprChildNode(ast, TokenTypes.NOT_EQUAL);
+        boolean error = false;
+        if (isNullTest(outerNotEqualNode)) {
+            final DetailAST nestedIfNode = getNestedIfNode(ast);
+            if (nestedIfNode != null) {
+                final DetailAST instanceOfNode =
+                        getExprChildNode(nestedIfNode, TokenTypes.LITERAL_INSTANCEOF);
+                if (instanceOfNode != null) {
+                    error = isTextEqual(instanceOfNode, outerNotEqualNode);
+                }
+            }
+        }
+        return error;
+    }
+
+    private static boolean checkOuterInstanceOf(DetailAST ast) {
+        final DetailAST outerInstanceOfNode =
+                getExprChildNode(ast, TokenTypes.LITERAL_INSTANCEOF);
+        boolean error = false;
+        if (outerInstanceOfNode != null) {
+            final DetailAST nestedIfNode = getNestedIfNode(ast);
+            if (nestedIfNode != null) {
+                final DetailAST notEqualNode =
+                        getExprChildNode(nestedIfNode, TokenTypes.NOT_EQUAL);
+                if (isNullTest(notEqualNode)) {
+                    error = isTextEqual(outerInstanceOfNode, notEqualNode);
+                }
+            }
+        }
+        return error;
+    }
+
+    /**
+     * To get specific Node that is child of EXPR Node.
+     *
+     * @param ast DetailAST
+     * @param token int
+     * @return DetailAST
+     */
+    private static DetailAST getExprChildNode(DetailAST ast, int token) {
+        return ast.findFirstToken(TokenTypes.EXPR)
+                .findFirstToken(token);
+    }
+
+    /**
+     * Judge whether this is a null test.
```

Comment:
it is hard to understand what function does by its javadoc, could you please add more description?

Reply:
done